playwright/src/fixtures/MHmodal.ts

```diff
@@ -23,6 +29,101 @@ class MHModal {
     // Wait for the highlights to be loaded in MH modal
     await this.page.waitForSelector('[class*=HighlightOuterWrapper]')
   }
+
+  async chapterDropdownCount() {
+    // Total number of checkboxes in the chapter dropdown
+    const Toc = new TOC(this.page)
+    const chapterDropdownCount = (await Toc.chapterCount()) + (await Toc.standalonePagesCount())
+    return chapterDropdownCount
+  }
+
+  async toggleChapterDropdown() {
+    // Open the chapter dropdown filter
+    await this.chapterDropdownLocator.click()
+  }
+
+  async toggleColorDropdown() {
+    // Open the color dropdown filter
+    await this.colorDropdownLocator.click()
+  }
+
+  async colorNumber(color: string) {
+    // Assign number to each color in the color dropdown
+    if (color === 'yellow') {
+      return '0'
+    } else if (color === 'green') {
+      return '1'
+    } else if (color === 'blue') {
+      return '2'
+    } else if (color === 'purple') {
+      return '3'
+    } else if (color === 'pink') {
+      return '4'
+    }
+  }
+
+  async checkboxChecked(n: any) {
+    // verify the checkbox status is checked in the chapter/color dropdown
+    if (typeof n === 'string') {
+      n = await this.colorNumber(n)
+    }
+    const checkBoxHtml = await this.checkBoxStatus.nth(await n).innerHTML()
+    if (checkBoxHtml.includes('type="checkbox" checked')) {
+      return true
+    }
+  }
+
+  async checkboxDisabled(n: any) {
+    // verify the checkbox status is disabled in the chapter/color dropdown
+    if (typeof n === 'string') {
+      n = await this.colorNumber(n)
+    }
+    const checkBoxHtml = await this.checkBoxStatus.nth(n).innerHTML()
```

Comment:
`nth` should receive a number but `colorNumber` returns strings

Reply:
removed the quotes